runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -413,8 +299,34 @@ private void removeCounter()
 		counter = null;
 	}
 
-	private static WorldArea buildCannonWorldArea(WorldPoint worldPoint)
+	private WorldPoint getCannonWorldPoint(int var)
+	{
+		if (var == -1)
+		{
+			return null;
+		}
+
+		// See VarPlayer#DWARF_CANNON_LOCATION
+		int x = (var % 268_435_456) / 16_384;
+		int y = var % 16_384;
+		int z = var / 268_435_456;
+
+		return new WorldPoint(x, y, z);
+	}
+
+	private void updateVars()
 	{
-		return new WorldArea(worldPoint.getX() - 1, worldPoint.getY() - 1, 3, 3, worldPoint.getPlane());
+		VarbitChanged vc = new VarbitChanged();
+		ArrayList<Integer> varps = new ArrayList<>();
+
+		varps.add(VarPlayer.DWARF_CANNON_LOCATION.getId());
+		varps.add(VarPlayer.DWARF_CANNON_PARTS_ASSEMBLED.getId());
+
+		for (int varp : varps)
+		{
+			vc.setVarpId(varp);
+			vc.setValue(client.getVarpValue(varp));
+			onVarbitChanged(vc);
+		}
```

Comment:
This seems... highly questionable. What is the purpose to manually calling `onVarbitChanged()` on plugin startup? It's extremely bad form to be calling an event subscriber directly outside of tests.